pdf/pdfselection.cpp

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright (C) 2015 Caliste Damien.
+ * Contact: Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "pdfselection.h"
+#include "pdfdocument.h"
+
+class PDFSelection::Private
+{
+public:
+    Private(): canvas(nullptr) {
+        roles.insert( Rect, "rect" );
+        roles.insert( Text, "text" );
+
+        iPageStart = iPageStop = -1;
+        iBoxStart = iBoxStop = -1;
+
+        wiggle = 4.;
```

Comment:
For the roles being initialised here, I did it after a comment in a previous PR, see https://github.com/sailfishos/sailfish-office/pull/36#discussion_r35949539 but reading again today, I notice that you already commented at that time that it can be put in the accessor of roles.

The i stand for index, will change this.